bin/p2-exec/main_darwin.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	// #include <stdlib.h>
+	// #include <unistd.h>
+	// #include <sys/resource.h>
+	// #include <sys/types.h>
+	// #include <sys/syslimits.h>
+	// #include <pwd.h>
+	// #include <grp.h>
+	"C"
+	"unsafe"
+
+	"github.com/square/p2/pkg/user"
+)
+
+func nolimit() error {
+	maxFDs := C.OPEN_MAX
```

Comment:
OPEN_MAX is usually set to some unreasonably low value. Try something like this instead: https://github.com/square/prodeng/blob/master/nolimit/nolimit.c#L65-L75